### ISSUE TITLE
VS2012.4 update fails when a newer update is already installed

### ISSUE THREAD

**Issue by dachew** [daptiv/visualstudio#40]
I already have VS2012.4 installed, but the latest cookbook doesn't seem to handle this situation very well.

Error info:
```bash
==> default: ================================================================================
==> default: Error executing action `install` on resource 'windows_package[Visual Studio 2012 Update 4 (KB2707250)]'
==> default: ================================================================================
==> default:
==> default: Mixlib::ShellOut::ShellCommandFailed
==> default: ------------------------------------
==> default: Expected process to exit with [0, 127, 3010], but received '-2147205120'
==> default: ---- Begin output of start "" /wait "d:\chef\cache\d88c9d4d8a481dbe45c75de7f347e9f6\VS2012.4.exe" /Q /norestart /noweb /Log "C:\Program Files (x86)\Microsoft Visual Studio 11.0\vsinstallupdate.log" & exit %%ERRORLEVEL%% ----
==> default: STDOUT:
==> default: STDERR:
==> default: ---- End output of start "" /wait "d:\chef\cache\d88c9d4d8a481dbe45c75de7f347e9f6\VS2012.4.exe" /Q /norestart /noweb /Log "C:\Program Files (x86)\Microsoft Visual Studio 11.0\vsinstallupdate.log" & exit %%ERRORLEVEL%% ----
==> default: Ran start "" /wait "d:\chef\cache\d88c9d4d8a481dbe45c75de7f347e9f6\VS2012.4.exe" /Q /norestart /noweb /Log "C:\Program Files (x86)\Microsoft Visual Studio 11.0\vsinstallupdate.log" & exit %%ERRORLEVEL%% returned -2147205120
```

And a snippet from the vs2012.4.exe install log:
```
[10E0:10DC][2015-11-17T11:32:20]: Calculating patch applicability for target product code: {FEB375AB-6EEC-3929-8FAF-188ED81DD8B5}, context: Machine
[10E0:10DC][2015-11-17T11:32:20]: Detected related bundle: {17c2e197-cf26-443b-8beb-53151940df3f}, type: Dependent, scope: PerMachine, version: 11.0.50727.1, operation: None
[10E0:10DC][2015-11-17T11:32:20]: MUX:  Detected related bundle. Installing bundle Lcid:1033  related bundle tag: vs_professional,1033
[10E0:10DC][2015-11-17T11:32:20]: MUX:  Detected Related Bundle called with ProductCode: {17c2e197-cf26-443b-8beb-53151940df3f}
[10E0:10DC][2015-11-17T11:32:20]: MUX:  Bundle Tag: PackageId=vs_professional, lcid=1033, IsLanguagePack=False
[10E0:10DC][2015-11-17T11:32:20]: MUX:  Request adding Language Pack to list: Bundle Id={17c2e197-cf26-443b-8beb-53151940df3f}, Bundle Lcid=1033, Tag=vs_professional,1033, Action=Install
[10E0:10DC][2015-11-17T11:32:20]: MUX:  Bundle Tag: PackageId=vs_professional, lcid=1033, IsLanguagePack=False
[10E0:10DC][2015-11-17T11:32:20]: Detected related bundle: {6d052d71-b953-48cd-8a75-3462b00efeb7}, type: Upgrade, scope: PerMachine, version: 11.0.61219.0, operation: Downgrade
[10E0:10DC][2015-11-17T11:32:20]: MUX:  Detected related bundle. Installing bundle Lcid:1033  related bundle tag: vsupdate_KB2707250,1033
[10E0:10DC][2015-11-17T11:32:20]: MUX:  A higher version bundle has already been installed.
[10E0:10DC][2015-11-17T11:32:20]: Error 0x80070642: UX aborted detect related bundle.
[10E0:10DC][2015-11-17T11:32:20]: Error 0x80070642: Failed to detect related bundles in HKLM for non-elevated process
[10E0:10DC][2015-11-17T11:32:20]: Error 0x80070642: Failed to detect bundles.
[10E0:10DC][2015-11-17T11:32:20]: Condition 'NOT (CurrentOperation = "Uninstall")' evaluates to true.
[10E0:10DC][2015-11-17T11:32:20]: Condition 'NOT (CurrentOperation = "Uninstall")' evaluates to true.
[10E0:10DC][2015-11-17T11:32:20]: Condition 'NOT (CurrentOperation = "Uninstall")' evaluates to true.
[10E0:10DC][2015-11-17T11:32:20]: Condition 'NOT (CurrentOperation = "Uninstall")' evaluates to true.
[10E0:10DC][2015-11-17T11:32:20]: Condition 'NOT (CurrentOperation = "Uninstall")' evaluates to true.
[10E0:10DC][2015-11-17T11:32:20]: Condition 'NOT (CurrentOperation = "Uninstall")' evaluates to true.
[10E0:10DC][2015-11-17T11:32:20]: Condition 'NOT (CurrentOperation = "Uninstall")' evaluates to true.
[10E0:10DC][2015-11-17T11:32:20]: MUX:  Stop Block: HigherVersionBlock : The product version that you are trying to set up is earlier than the version already installed on this computer.
[10E0:10DC][2015-11-17T11:32:20]: MUX:  Stop Block: NoApplicableUpdates : Update 4 does not apply, or is blocked by another condition on your computer.
[10E0:10DC][2015-11-17T11:32:20]: MUX:  Hit a stop blocker, exiting now
[10E0:10DC][2015-11-17T11:32:20]: Detect complete, result: 0x80070642
[10E0:13FC][2015-11-17T11:32:20]: MUX:  Exit Run
[10E0:13FC][2015-11-17T11:32:20]: MUX:  Metrics: ShouldSendData=False
[10E0:13FC][2015-11-17T11:32:20]: MUX:  Permission to upload: No
[10E0:13FC][2015-11-17T11:32:20]: MUX:  Preparing to serialize data.
[10E0:13FC][2015-11-17T11:32:20]: MUX:  Data serialized.
[10E0:13FC][2015-11-17T11:32:20]: MUX:  Number of SQM File queued: 0
[10E0:13FC][2015-11-17T11:32:20]: MUX:  SQM sent: True
[10E0:13FC][2015-11-17T11:32:20]: Setting string variable 'CEIPConsent' to value ''
[10E0:13FC][2015-11-17T11:32:20]: Setting string variable 'SqmOption' to value ''
[10E0:13FC][2015-11-17T11:32:20]: MUX:  Metrics: Configuration State
[10E0:13FC][2015-11-17T11:32:20]: MUX:  ----------------------------
```

**Comment by sneal** [comment]
@dachew Do you have SP5 already installed?

**Comment by dachew** [comment]
sounds good.  :shipit:

**Comment by sneal** [comment]
Fixed by #41
